File: smart-fridge-java-backend/src/main/java/css/ch/smartfridgejavabackend/auth/service/AuthServiceImpl.java
package css.ch.smartfridgejavabackend.auth.service;

import css.ch.smartfridgejavabackend.user.model.User;
import css.ch.smartfridgejavabackend.user.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class AuthServiceImpl implements AuthService {
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;

    @Override
    public String authenticate(String email, String rawPassword) {
        User user = userRepository.findUserByEmail(email)
                .orElseThrow(() -> new BadCredentialsException("Ungültige E-Mail"));

        if (!passwordEncoder.matches(rawPassword, user.getPassword())) {
            throw new BadCredentialsException("Falsches Passwort");
        }

        return user.getId();
    }
}

File: smart-fridge-backend/src/main/kotlin/ch/bbzw/smartfridgebackend/auth/service/AuthServiceImpl.kt
package ch.bbzw.smartfridgebackend.auth.service

import ch.bbzw.smartfridgebackend.user.repository.UserRepository
import lombok.RequiredArgsConstructor
import org.springframework.security.authentication.BadCredentialsException
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service

@Service
@RequiredArgsConstructor
class AuthServiceImpl(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder
) : AuthService {

    override fun authenticate(email: String, rawPassword: String): String? {
        val user = userRepository.findUserByEmail(email).orElseThrow { BadCredentialsException("Ungültige E-Mail") }

        if (!passwordEncoder.matches(rawPassword, user.password)) {
            throw BadCredentialsException("Falsches Passwort")
        }

        return user.id
    }
}